docs/configuration.md

```diff
@@ -0,0 +1,90 @@
+<!--
+    Copyright 2019 VMware, Inc.
+    SPDX-License-Identifier: BSD-2-Clause
+-->
+
+# Precaution configuration
+
+Precaution provides you with the option to use optional `.precaution.json` file.
+This file **should** be in the root directory of your project.
+
+All paths provided in the options are processed as if the
+they are from the root directory.
+
+This file can be configured with the following options:
+
+| Option | Syntax |
+|---|---|
+| [Exclude](#exclude) | [Globs](#globs) |
+
+## Exclude
+
+This option can be used to tell Precaution to exclude files or folders from processing.
+
+If you want to exclude a specific file from the Precaution scan you can provide
+the full path to it starting from the root directory of your project.
+Then if you want to ignore `src/cmd/main.go` file you will have this `.precaution.json` file:
+```
+{
+    exclude: ["src/cmd/main.go"]
+}
+```
+
+For example, if you are using Golang and your dependencies are in
+a `vendor/` folder then you can tell Precaution to skip all files
+within the `vendor/` folder with this `.precaution.json` file:
+```
+{
+    exclude: ["vendor/**"]
+}
+```
+
+You can use the same syntax if you are using Python
+and you want to exclude all test files in your `test/` folder.
+
+Another possible scenario is when you want to exclude all Golang test files.
+You can achieve this with this `.precaution.json` file:
+```
+{
+    exclude: ["*_test.go"]
+}
+```
+
+The exclude option in the `.precaution.json` file supports multiple exclude rules:
+```
+{
+    exclude: ["*_test.go", "vendor/**", "src/cmd/main.go"]
+}
+```
+
+## Globs
+
+Some of the Precaution configuration options are using Globs as a syntax.
+
+These are just some of the example patterns we are supporting.
```

Comment:
There are more patterns we support? How does the user find out what they all are?

Reply:
I mentioned:
- Brace Expansion
- Extended glob matching
below too because in the minimatch docs they say those are supported too.

Reply:
Agree, this statement doesn't add value.

Reply:
Okay, I will remove it.